src/ucp/tag/tag_match.h

```diff
@@ -45,9 +45,12 @@ typedef struct {
 /**
  * Hash table entry for tag message fragments
  */
-typedef union {
-    ucs_queue_head_t      unexp_q;    /* Queue of unexpected descriptors */
-    ucp_request_t         *exp_req;   /* Expected request */
+typedef struct {
+    union {
+        ucs_queue_head_t  unexp_q;    /* Queue of unexpected descriptors */
+        ucp_request_t     *exp_req;   /* Expected request */
+    };
+    size_t                offset;
```

Comment:
why do we need to keep offset in the "frag"? it's already part of header
can we take offset from last entry in unexp_q? or recv request state if it's "exp_req"?